metricshub-engine/src/main/java/org/metricshub/engine/strategy/source/compute/ComputeProcessor.java

```diff
@@ -610,6 +669,70 @@ public void process(@SpanAttribute("compute.definition") final DuplicateColumn d
 		sourceTable.setRawData(SourceTable.tableToCsv(sourceTable.getTable(), TABLE_SEP, false));
 	}
 
+	@Override
+	@WithSpan("Compute Encode Exec")
+	public void process(@SpanAttribute("compute.definition") final Encode encode) {
+		if (encode == null) {
+			log.warn("Hostname {} - Encode object is null, the table remains unchanged.", hostname);
+			return;
+		}
+
+		if (encode.getColumn() == null || encode.getColumn() == 0) {
+			log.warn("Hostname {} - The column index in Encode cannot be null or 0, the table remains unchanged.", hostname);
+			return;
+		}
+
+		if (encode.getProtocol() == null || encode.getProtocol().isEmpty()) {
+			log.warn("Hostname {} - The protocol in Encode cannot be null or empty, the table remains unchanged.", hostname);
+			return;
+		}
+
+		final int columnIndex = encode.getColumn() - 1;
+		final String protocol = encode.getProtocol();
+
+		for (final List<String> line : sourceTable.getTable()) {
+			if (columnIndex < line.size()) {
+				final String valueToBeEncoded = line.get(columnIndex);
+
+				final String newValue;
+
+				switch (protocol.toLowerCase()) {
+					case "base64":
+						newValue = Base64.getEncoder().encodeToString((valueToBeEncoded.trim()).getBytes(StandardCharsets.UTF_8));
+						break;
+					case "urlencode":
+						newValue = URLEncoder.encode(valueToBeEncoded, StandardCharsets.UTF_8);
+						break;
+					case "md5":
```

Comment:
This is not encoding, this is hashing (and it cannot be decoded!). This should be in a separate compute class: `Hash.java`.